packages/lesswrong/server/callbacks/localgroupCallbacks.ts

```diff
@@ -0,0 +1,16 @@
+import { getCollectionHooks } from '../mutationCallbacks';
+
+
+getCollectionHooks("Localgroups").createValidate.add(async (validationErrors: Array<any>, {document: group}: {document: DbLocalgroup}) => {
+  if (!group.isOnline && !group.location)
+    throw new Error("Location is required for local groups");
+  
+  return validationErrors;
+});
+
+getCollectionHooks("Localgroups").updateValidate.add(async (validationErrors: Array<any>, {oldDocument, newDocument}: {oldDocument: DbLocalgroup, newDocument: DbLocalgroup}) => {
+  if (!newDocument.isOnline && !newDocument.location)
+    throw new Error("Location is required for local groups");
```

Comment:
This is how I handled the location requirement. I'm not sure if this error message is clear though - if you have any suggestions for how to refer to a non-online group besides "local group" let me know.